test/UnitTests/Microsoft.Testing.Platform.UnitTests/CommandLine/PlatformCommandLineProviderTests.cs

```diff
@@ -0,0 +1,169 @@
+        CommandLineOption option = provider.GetCommandLineOptions().First(x => x.Name == PlatformCommandLineProvider.PortOptionKey);
+
+        ValidationResult validateOptionsResult = await provider.ValidateOptionArgumentsAsync(option, [timeout]).ConfigureAwait(false);
+        Assert.AreEqual(isValid, validateOptionsResult.IsValid);
+
+        if (!isValid)
+        {
+            Assert.AreEqual(string.Format(CultureInfo.InvariantCulture, PlatformResources.PlatformCommandLinePortOptionSingleArgument, PlatformCommandLineProvider.PortOptionKey), validateOptionsResult.ErrorMessage);
+        }
+    }
+
+    [Arguments("32", true)]
+    [Arguments("32.32", false)]
+    [Arguments("invalid", false)]
+    public async Task IsValid_If_ClientPort_Is_Integer(string timeout, bool isValid)
+    {
+        var provider = new PlatformCommandLineProvider();
+        CommandLineOption option = provider.GetCommandLineOptions().First(x => x.Name == PlatformCommandLineProvider.ClientPortOptionKey);
+
+        ValidationResult validateOptionsResult = await provider.ValidateOptionArgumentsAsync(option, [timeout]).ConfigureAwait(false);
+        Assert.AreEqual(isValid, validateOptionsResult.IsValid);
+
+        if (!isValid)
+        {
+            Assert.AreEqual(string.Format(CultureInfo.InvariantCulture, PlatformResources.PlatformCommandLinePortOptionSingleArgument, PlatformCommandLineProvider.ClientPortOptionKey), validateOptionsResult.ErrorMessage);
+        }
+    }
+
+    [Arguments("32", true)]
+    [Arguments("32.32", false)]
+    [Arguments("invalid", false)]
+    public async Task IsValid_If_ExitOnProcessExit_Is_Integer(string timeout, bool isValid)
+    {
+        var provider = new PlatformCommandLineProvider();
+        CommandLineOption option = provider.GetCommandLineOptions().First(x => x.Name == PlatformCommandLineProvider.ExitOnProcessExitOptionKey);
+
+        ValidationResult validateOptionsResult = await provider.ValidateOptionArgumentsAsync(option, [timeout]).ConfigureAwait(false);
+        Assert.AreEqual(isValid, validateOptionsResult.IsValid);
+
+        if (!isValid)
+        {
+            Assert.AreEqual(string.Format(CultureInfo.InvariantCulture, PlatformResources.PlatformCommandLineExitOnProcessExitSingleArgument, PlatformCommandLineProvider.ExitOnProcessExitOptionKey), validateOptionsResult.ErrorMessage);
+        }
+    }
+
+    public async Task IsValid_If_Diagnostics_Provided_With_Other_Diagnostics_Provided()
+    {
+        var provider = new PlatformCommandLineProvider();
+        var options = new Mock<ICommandLineOptions>();
+        _ = options.Setup(o => o.IsOptionSet(PlatformCommandLineProvider.DiagnosticOptionKey)).Returns(true);
+        _ = options.Setup(o => o.IsOptionSet(PlatformCommandLineProvider.DiagnosticOutputDirectoryOptionKey)).Returns(true);
+        _ = options.Setup(o => o.IsOptionSet(PlatformCommandLineProvider.DiagnosticOutputFilePrefixOptionKey)).Returns(true);
```

Comment:
consider: testing this without Moq. If you stick with moq, consider running validate on the mock.

(It might be just the horror story I am living through right now on testfx to fix moq based tests).

Reply:
You can also use the strict mode by default (in the ctor of the mock) where it will fail if a call is made to some non-mocked API.

Reply:
removed mock,